### PR TITLE
WIP: Feature/no reachable code

### PR DESCRIPTION
**Pull request by JayaKrishnaNamburu**
Currently covering the `BlockStatements`, still there are a few cases where i am missing 😅. Ready to refactor with the discussion for the changes 😄

### CODE REVIEW

**Comment by sebmck** [comment]
I think a better approach is to have the operations be performed on the `BlockStatement` itself. That way you can implement an autofix that removes all the code. I think there'd also be a way to use `getCompletionRecords` #131 to make this reliably handle conditionals that always result in a completion.

File: packages/@romejs/js-compiler/transforms/lint/unReachableCode.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+    if (node.type === 'BlockStatement') {
+      const records = getCompletionRecords(node);
+      for (const record of records) {
+        if (
+          record.type === 'COMPLETION' &&
+          node.body[node.body.length - 1].type !== 'ReturnStatement'
+        ) {
+          context.addNodeDiagnostic(node.body[node.body.length - 1], {
+            category: 'lint/unRecheableCode',
```
Comment (sebmck) [inline packages/@romejs/js-compiler/transforms/lint/unReachableCode.ts]: Typo

```suggestion
            category: 'lint/unReachableCode',
```

File: packages/@romejs/js-compiler/transforms/lint/unReachableCode.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+            message: 'Unrecheable code',
```
Comment (sebmck) [inline packages/@romejs/js-compiler/transforms/lint/unReachableCode.ts]:
```suggestion
            message: 'Unreachable code',
```

File: packages/@romejs/js-compiler/transforms/lint/unReachableCode.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  name: 'unReachableCode',
+  enter(path: Path): AnyNode {
+    const {node, context} = path;
+
+    if (node.type === 'BlockStatement') {
+      const records = getCompletionRecords(node);
+      for (const record of records) {
+        if (
+          record.type === 'COMPLETION' &&
+          node.body[node.body.length - 1].type !== 'ReturnStatement'
```
Comment (sebmck) [inline packages/@romejs/js-compiler/transforms/lint/unReachableCode.ts]: Why do we only care about the last node?

Comment (JayaKrishnaNamburu) [reply packages/@romejs/js-compiler/transforms/lint/unReachableCode.ts]: I am working on this, became a little tricky for me and ended up confusing. When i am running on `./scripts/dev-rome lint` linting is failing.  I will will be working on this and share a update soon 😅

**Comment by sebmck** [comment]
Hey @JayaKrishnaNamburu! Wanted to check in and see if there's anything I can do to help with this PR.

**Comment by JayaKrishnaNamburu** [comment]
Hey @sebmck sorry for the delay, was caught up with something else. I will try to complete it and share an update :) Thank you

**Comment by sebmck** [comment]
Closing this out due to inactivity. If you're able to pick this up again, please feel free to open another PR! Thank you!

**Pull request closed without merging**